pkgs/tools/inputmethods/ibus-engines/ibus-typing-booster/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, fetchFromGitHub, autoreconfHook, python3, ibus, pkgconfig, gtk3, makeWrapper, m17n_lib
+, pango, gdk_pixbuf, atk, wrapGAppsHook
+}:
+
+let
+
+  python = python3.withPackages (ps: with ps; [
+    pygobject3
+    dbus-python
+  ]);
+
+  mkRepoPaths = stdenv.lib.makeSearchPath "lib/girepository-1.0";
+
+in
+
+stdenv.mkDerivation rec {
+  name = "ibus-typing-booster-${version}";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "mike-fabian";
+    repo = "ibus-typing-booster";
+    rev = version;
+    sha256 = "01kpxplk9nh56f32fkq3nnsqykbzpi7pcxbfp38dq0prgrhw9a6b";
+  };
+
+  nativeBuildInputs = [ autoreconfHook pkgconfig makeWrapper wrapGAppsHook ];
+  buildInputs = [ python ibus gtk3 m17n_lib ];
+
+  preBuild = ''
+    export GI_TYPELIB_PATH="${mkRepoPaths [ gtk3 pango.out gdk_pixbuf atk ibus ]}"
```

Comment:
Add `gobjectIntrospection` to `nativeBuildInputs` to populate the variable automatically.